Core/WebCacheManager.swift

```diff
@@ -101,62 +64,60 @@ public class WebCacheManager {
+
             DispatchQueue.global(qos: .background).async {
                 _ = group.wait(timeout: .now() + 5)
                 DispatchQueue.main.async {
                     completion()
                 }
             }
         }
-        
-    }
 
-    /// The Fire Button does not delete the user's DuckDuckGo search settings, which are saved as cookies. Removing these cookies would reset them and have undesired
-    ///  consequences, i.e. changing the theme, default language, etc.  These cookies are not stored in a personally identifiable way. For example, the large size setting
-    ///  is stored as 's=l.' More info in https://duckduckgo.com/privacy
-    private func isDuckDuckGoOrAllowedDomain(cookie: HTTPCookie, domain: String) -> Bool {
-        return cookie.domain == domain || (cookie.domain.hasPrefix(".") && domain.hasSuffix(cookie.domain))
     }
 
     public func clear(dataStore: WebCacheManagerDataStore = WKWebsiteDataStore.default(),
-                      appCookieStorage: CookieStorage = CookieStorage(),
                       logins: PreserveLogins = PreserveLogins.shared,
                       completion: @escaping () -> Void) {
-        extractAllowedCookies(from: dataStore.cookieStore, cookieStorage: appCookieStorage, logins: logins) {
-            self.clearAllData(dataStore: dataStore, completion: completion)
-        }
-    }
 
-    private func clearAllData(dataStore: WebCacheManagerDataStore, completion: @escaping () -> Void) {
-        dataStore.removeAllData(completion: completion)
-    }
-    
-    private func extractAllowedCookies(from cookieStore: WebCacheManagerCookieStore?,
-                                       cookieStorage: CookieStorage,
-                                       logins: PreserveLogins,
-                                       completion: @escaping () -> Void) {
-        
-        guard let cookieStore = cookieStore else {
-            completion()
-            return
-        }
-        
-        cookieStore.getAllCookies { cookies in
-            for cookie in cookies {
-                if cookie.domain == Constants.cookieDomain || logins.isAllowed(cookieDomain: cookie.domain) {
-                    cookieStorage.setCookie(cookie)
+        dataStore.removeAllDataExceptCookies {
+            guard let cookieStore = dataStore.cookieStore else {
+                completion()
+                return
+            }
+
+            let group = DispatchGroup()
+
+            cookieStore.getAllCookies { cookies in
+                let cookiesToRemove = cookies.filter { !logins.isAllowed(cookieDomain: $0.domain) && $0.domain != Constants.cookieDomain }
+
+                for cookie in cookiesToRemove {
+                    group.enter()
+                    cookieStore.delete(cookie) {
+                        group.leave()
+                    }
+                }
+            }
+
+            DispatchQueue.global(qos: .background).async {
+                _ = group.wait(timeout: .now() + 5)
```

Comment:
Maybe we should fire debug pixel when it's timing out?

Reply:
Another question: is it possible that this executes before 'getAllCookies' returns (and before first 'enter' is called on a group)?

Reply:
Yeah. probably want an `enter()` before calling `getAllCookies` and a leave inside it, assuming that the completion handler for `getAllCookies` always gets called.  If it doesn't ... not sure what you could do here.

Reply:
I like that suggestion, moving the `DispatchGroup` handling into the `getAllCookies` callback sounds good  – done.